### ISSUE TITLE
Add Block: Logo

### ISSUE THREAD

**Issue by mtias** [WordPress/gutenberg#1851]
It'd be nice to have a block that just renders the custom logo as registered with `add_theme_support`.

**Comment by karmatosed** [comment]
Here is a suggestion of how that could look design wise. I think it's useful to give option to align and change it right there.

![logo](https://user-images.githubusercontent.com/253067/28088212-e440c602-667c-11e7-9ce9-1870bf6d490f.png)

**Comment by paaljoachim** [comment]
How would we go about adjusting the size of the logo?

**Comment by karmatosed** [comment]
I am torn about that. The theme sets the size, I think having it kept to that size makes sense. At least it should be the first version and then perhaps testing will show if it needs to change. This isn't placing an image, it's the site logo.

**Comment by westonruter** [comment]
@mtias Note that `add_theme_support` registers not the custom logo itself, but rather the ability for the user to assign a custom logo in the Customizer. So this is _not_ currently including the ability to _change_ the custom logo as assigned? If not, then this would be directly related to #1224 and being able to source block attributes from global settings.

Also, an endpoint for updating the `custom_logo` theme mod is currently planned for as part of https://github.com/WP-API/wp-api-customize-endpoints

On the other hand, if the block is just going to be read-only, then really this can be implemented using server-side rendering of a block (#780), re-using a block's PHP rendering logic.

**Comment by mtias** [comment]
@westonruter we can start with just display, and then add the necessary pieces for updating the data as well. We'll need a coherent system for specifying attributes that map to options, fields, user meta, etc, etc.

**Comment by melchoyce** [comment]
@karmatosed Rather than showing the logo in the Inspector with instructions for editing, I think we can rely on having an "edit" icon when the block itself is selected, similar to the image block. Like the image block, we might also want to include a link option:

![image](https://user-images.githubusercontent.com/2846578/28221310-d14a8a68-6890-11e7-9a4e-9e1c972ec28f.png)

**Comment by westonruter** [comment]
Should not the block interface actually be identical to the image block in general? The key difference would be in the attribute storage, where the selected image is stored globally in the theme mod as opposed locally in the content. In that way, it should probably get the styling treatment as lifted/external/global blocks: #1516.

**Comment by StaggerLeee** [comment]
There is no way Page "template" style and design will match rest of theme, be it custom theme or from repository.
This thing with logo in Posts/Pages is pointless, unless you have plans to go 100% front-end theme editor way, visual theme designer.

**Comment by WPDevHQ** [comment]
I think this as a block is very fitting for future vision. Maybe these should have been called modules rather than blocks? :P 

Some of us theme developers a looking at Gutenberg beyond just a page/post editor - we are looking at the possibility of creating sections/panels for global assignments into places like the header.php, footer.php and even sidebar.php

We are looking at the possibility of pulling in things like the menu, an entire sidebar/widget area and not just an individual widget to build out these global sections/panels - so the blocks that can be brought in by core the better as it will save themes/plugins having to hack them in!

**Comment by melchoyce** [comment]
Hey @WPDevHQ! That's the long-term goal, just going to take a couple iterations to get there :)

**Comment by karmatosed** [comment]
As the project moves towards merge proposal, we are moving issues that aren't needed for that to projects. This doesn't mean they won't get done, they totally can and that's why we're moving to projects. This allows us to focus on the issues that have to happen to Gutenberg. I am closing this but it will live in projects.

**Comment by karmatosed** [comment]
This is starting to be explored by @pablinos

**Comment by pablinos** [comment]
Thanks @karmatosed! The idea of this block came up during some discussions about full site editing. We were wondering what the future looks like for the logo theme option, and what would make this different from some kind of reusable image block?

The main advantage would seem to be backwards compatibility with themes, but would it be worthwhile to fallback to another mechanism for themes (now or in the future) that don't have support for logos?

Also, until we have the concept of a page header that we can add blocks to, how useful is this block? I can see circumstances where someone might want to include their logo in a sidebar or main post content, so maybe that's enough reason to build it.

**Comment by strarsis** [comment]
The logo set in theme can also be treated as a site-wide logo, setting site app icon, favicon, etc.

**Comment by nickcernis** [comment]
Instead of a dedicated logo block, should the `wp:site-title` block be extended to offer logo properties? `wp:site-title` would then:

- Show the site title as text by default.
- Provide controls to add a logo via the block properties.
- Offer a “hide site title text” option so users can show the logo only if they wish.
- Potentially use the user's `custom-logo` by default if one is set. Themes with `custom-logo` support that add `<!-- wp:site-title /-->` in a template part will then continue to show the `custom-logo`.

The logo is so often used as well as the title or in place of it, so this feels like a logical evolution.

**Comment by strarsis** [comment]
@nickcernis: Though logo and title are separate in traditional WordPress themes.

**Comment by pablinos** [comment]
This has been brought up before, that they should be in the same block, but you then end up with the need to add more alignment settings etc. 

Another option is to have separate `site-title` and `site-logo` blocks, and then to offer various block variations to cover how they would be used together.

**Comment by nickcernis** [comment]
> Another option is to have separate site-title and site-logo blocks, and then to offer various block variations to cover how they would be used together.

This sounds good. I like that it avoids the need to have “show title above/below logo” controls. We should probably favour drag-and-drop placement over placement controls where we can.

**Comment by strarsis** [comment]
What can I use right now?
For the layout of a new site design the new full page editing feature comes right in time!
The menu block is already available, now I only need the logo block to rebuild the whole site as blocks.

**Comment by pablinos** [comment]
This is the logo block PR #18811 - It's currently held up by a couple of problems with getting options from the API when you're not an admin. You might want to try rebasing that and installing it as a block.

You could always use an image of course!

**Comment by scruffian** [comment]
We could still merge this and have it working like the other blocks in FSE...

**Comment by mtias** [comment]
Agreed, I don't think the admin circumstance should be a blocker. @epiqueras curious what you think of this block? I am not entirely sure if the logo is automatically being used for other purposes (favicon, mobile icon, etc).

**Comment by epiqueras** [comment]
Yeah, that's my question, too, because it seems much more straightforward to just put a regular Image block in your header template part. I don't see much point in a dedicated Site Logo block, especially if it's read-only.

**Comment by paaljoachim** [comment]
My guess is that Site Logo Block would be associated with the customizer. If one adds a site logo through Site Identity it would also automatically be added to the Site Logo block. That makes it different from a regular Image Block.

**Comment by shaunandrews** [comment]
I’d expect a Logo block to be different from a normal Image block in at least a few ways:

- No need for the Image caption in a Logo block. 
- I’d expect a logo block to support both Image and Text, and to smartly render the markup on the front end. 
- A Logo block could smartly adapt based on viewport size to hide and show variants of a logo.
- I’d expect to be able to use the Logo block to define my sites favicon. 

(It’s also possible I’m not understanding the discussion above. Feel free to ignore me if this is going off topic.)

**Comment by epiqueras** [comment]
Shouldn't the favicon be handled in a site setting somewhere? Most of the time the logo is a different asset than the favicon right?

**Comment by strarsis** [comment]
@epiqueras: Right, but for full site editing, all site features should be somehow modeled as Gutenberg blocks?

**Comment by epiqueras** [comment]
Only the ones that you would use in the front end. The asset used for the favicon is probably different from the one used by a template's header, for example.

**Comment by strarsis** [comment]
@epiqueras: The favicon/app icon can also be much more nuanced, with different icons (art direction) and postprocessing (background, margins, shadows) and app colors, see https://wordpress.org/plugins/favicon-by-realfavicongenerator/.

**Comment by epiqueras** [comment]
Yeah, that's more reason just to use a regular image block, right?

**Comment by paaljoachim** [comment]
Just a friendly reminder..:)
Could we get a status update of where this issue is at?

**Comment by scruffian** [comment]
This block has been merged, so I think we can close this issue.

**Comment by paaljoachim** [comment]
Closing issue.

**Comment by strarsis** [comment]
@scruffian, @paaljoachim: It would be great to have a new Gutenberg release so I can use the Logo block in production! 🐱

**Comment by paaljoachim** [comment]
Hey @strarsis 

The Logo block for the site editor (Full Site Editing) has been released here: https://github.com/WordPress/gutenberg/pull/18811

Regarding using various Full Site Editing blocks in the Post Editor (non FSE - regular Gutenberg screen) that I do not know anything about. Perhaps someone can reply to that aspect.
